src/Customer/Track/Events/Types/TrackCustomerEvent.hs

```diff
@@ -0,0 +1,44 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io/>
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+module Customer.Track.Events.Types.TrackCustomerEvent
+  ( module Customer.Track.Events.Types.TrackCustomerEvent
+  ) where
+
+import Customer.Aeson (defaultAesonOptions, mkObject, mkPair)
+import Data.Aeson (Object, ToJSON(toJSON), Value(..))
+import Data.Aeson.TH (deriveToJSON)
+import Data.Text (Text)
+
+data TrackCustomerEventBody = MkTrackCustomerEvent
+  { tceName      :: Text
+  , tceId        :: Maybe Text
+  , tceTimestamp :: Maybe Int
+  , tceData      :: Maybe CustomerEventData
+  }
+
+data CustomerEventData = MkCustomerEventData
```

Comment:
The [API docs for tracking](https://customer.io/docs/api/track/?region=eu#operation/track) are misleading. What they're actually documenting is an arbitrary JSON object, not a structured payload with those attributes.

`data` can contain arbitrary attributes, in addition to a few reserved ones for sending emails: `recipient`, `from_address`, and `reply_to`. The reserved attributes are optional.

Reply:
We capture this behavior in the ToJSON instance. All the additional fields are merging with description/type/properties into one JSON object.

Reply:
Sure, but `description`, `type`, and `properties` are not part of the API. They're *not* reserved fields. It's a mistake in the documentation.

Reply:
Oh, now I understand you